lib/network/modules/components/DirectionStrategy.js

```diff
@@ -221,9 +248,12 @@ class HorizontalStrategy extends DirectionInterface {
 
   /** @inheritdoc */
   sort(nodeArray) {
+    let idOrder = this.idPerLevel(nodeArray);
+
     nodeArray.sort(function(a, b) {
-      // Test on 'undefined' takes care of divergent behaviour in chrome
-      if (a.y === undefined || b.y === undefined) return 0;   // THIS HAPPENS
+      if (a.y === undefined || b.y === undefined) {
+        return idOrder[a.id] - idOrder[b.id];
```

Comment:
Could you use `return nodeArray.indexOf(a) - nodeArray.indexOf(b)` instead of creating idOrder?

Reply:
That was my first attempt. It doesn't work because the array get changed inline as the sorting runs.

So the indexing may change on every call to the custom `sort()` - and it does on chrome, even if there's no reason to change it. That's the quirk this issue tries to work around.

Basically, the fix is to register the initial sorting order and retain that by default.

Reply:
Why are the x or y values ever `undefined`?

Reply:
They come from the options. They are user-defined starting positions, not required. They can be `undefined` for that reason.

Reply:
I am somewhat uncomfortable with the idea of having to iterate this array twice in order to sort it.

Reply:
If the user specifies an undefined position, can we assign it a position, so that by the time it reaches this sort code, it'll never be undefined?

Reply:
What about just comparing `a.id` against `b.id` in the case that x or y is undefined?